### PR TITLE
threshold

### PR DESCRIPTION
**Pull request by julesbarbosa**
## What does this PR do?

set and get threshold info in model model 

## Changes

- [x] add / del threshold on clf models blob
- [x] get model info file

## Before reviewing

- [x] Are there any linked PRs?
- [ ] Did you test this PR?
- [ ] Is there any migration or seeding necessary?
- [ ] Did you add api to only one of the language (python/typescript)?

### Links:

-   xyme-backend: https://github.com/Accern/xyme-backend/pull/170
-  Xyme-notebooks: https://github.com/Accern/xyme-notebooks/pull/31/commits/a12c61f20e6f321689186fc83eaedbf736d081bd
### New API added that not implemented in other languages

If your answer is yes, please also update the todo_api.txt at where the API is missing.

### CODE REVIEW

File: packages/python/accern_xyme/accern_xyme.py [hunk 1]
```diff
@@ -659,6 +660,22 @@ def set_blob_owner(
...
+                "blob": blob_uri,
+                "threshold": threshold,
+            }))
+
+    def get_model_threshold(self, blob_uri: str) -> ModelThreshold:
+        return cast(ModelThreshold, self.request_json(
+            METHOD_GET, "/threshold", {
+                "blob": blob_uri,
+            }))
+
```
Comment (julesbarbosa) [inline packages/python/accern_xyme/accern_xyme.py:678]: delete

File: packages/python/accern_xyme/accern_xyme.py [hunk 2]
```diff
@@ -2274,6 +2291,20 @@ def set_owner(self, owner: NodeHandle) -> BlobOwner:
...
             }))
 
+    def set_model_threshold(self, threshold: float, pos_label: str) -> ModelInfo:
+        return cast(ModelInfo, self._client.request_json(
+            METHOD_PUT, "/threshold", {
+                "blob": self.get_uri(),
+                "threshold": threshold,
+                "pos_label": pos_label,
+            }))
+
```
Comment (julesbarbosa) [inline packages/python/accern_xyme/accern_xyme.py:2301]: remove model threshold -> sets both to none

File: packages/python/accern_xyme/accern_xyme.py (hunk 1 above)
Comment (julesbarbosa) [reply packages/python/accern_xyme/accern_xyme.py:678]: do typescript

File: packages/python/accern_xyme/accern_xyme.py [hunk 3]
```diff
@@ -2351,6 +2351,29 @@ def get_owner_node(self) -> str:
...
+    def del_model_threshold(self) -> ModelInfo:
+        return cast(ModelInfo, self._client.request_json(
+            METHOD_PUT, "/del_threshold", {
```
Comment (JosuaKrause) [inline packages/python/accern_xyme/accern_xyme.py:2367]: METHOD_DELETE, "/threshold"

File: packages/typescript/index.ts [hunk 4]
```diff
@@ -2627,6 +2627,42 @@ export class BlobHandle {
...
      * external-owned blob.
      */
+
+    public async setModelThreshold(
+        threshold: number,
+        pos_label: string
+    ): Promise<ModelInfo> {
+        return await this.client.requestJSON<ModelInfo>({
+            method: METHOD_PUT,
+            path: 'threshold',
```
Comment (TheAnkurGoswami) [inline packages/typescript/index.ts:2637]: `/threshold`

File: packages/typescript/index.ts [hunk 5]
```diff
@@ -2627,6 +2627,42 @@ export class BlobHandle {
...
+            args: {
+                blob: this.uri,
```
Comment (TheAnkurGoswami) [inline packages/typescript/index.ts:2639]: `this.getURI()`

**Review by TheAnkurGoswami — CHANGES_REQUESTED** [review]
minor changes

File: packages/typescript/index.ts [hunk 6]
```diff
@@ -2627,6 +2627,42 @@ export class BlobHandle {
...
+            },
+        });
+    }
+
+    public async delModelThreshold(): Promise<ModelInfo> {
+        return await this.client.requestJSON<ModelInfo>({
+            method: METHOD_DELETE,
+            path: '/threshold',
+            args: {
+                blob: this.uri,
```
Comment (julesbarbosa) [inline packages/typescript/index.ts:2651]: @TheAnkurGoswami is this not this.getUri() as well?

**Review by TheAnkurGoswami — DISMISSED** [review]
LGTM

File: packages/typescript/index.ts (hunk 6 above)
Comment (JosuaKrause) [reply packages/typescript/index.ts:2651]: please stay consistent -- use this.getURI here as well

Comment (julesbarbosa) [inline packages/typescript/index.ts:2651]:
```suggestion
                blob: this.getURI(),
```

**Review by JosuaKrause — APPROVED** [review]

**Pull request merged**